chia/server/rate_limits.py

```diff
@@ -81,25 +81,47 @@ def process_msg_and_check(
                 new_non_tx_count = self.non_tx_message_counts + 1
                 new_non_tx_size = self.non_tx_cumulative_size + len(message.data)
                 if new_non_tx_count > non_tx_freq * proportion_of_limit:
-                    return False
+                    return (
+                        False,
+                        f"non-tx count: {new_non_tx_count} "
+                        f"> {non_tx_freq * proportion_of_limit} "
+                        f"(scale factor: {proportion_of_limit})",
+                    )
                 if new_non_tx_size > non_tx_max_total_size * proportion_of_limit:
-                    return False
+                    return (
+                        False,
+                        f"non-tx size: {new_non_tx_size} "
+                        f"> {non_tx_max_total_size * proportion_of_limit}"
```

Comment:
Or perhaps a different code form to make this less likely to happen.  Maybe some function called like `f("non-tx-size", new_non_tx_size, non_tx_max_total_size, proportion_of_limit)`.  I think this could replace all these cases?

```suggestion
                        f"> {non_tx_max_total_size * proportion_of_limit} "
```

Or `" ".join([f"this", f"that", f"more")` perhaps.  Just trying to get rid of 1) adjacent string literal concatenation, 2) single line string literals spread over multiple lines, and 3) sequences of lines where some delineate separate parameters and others group as one parameter but without any indentation/parenthisization.

Reply:
"adjacent string literal concatenation" is that considered bad form?

Reply:
I personally find it error prone and confusing to read.  And I suppose I ought to look more at this since it's kinda broken up in two different ways.  Basically, I don't like looking through 'a list' and having some 'entries' be parametes and others be otherwise grouped without any indentation.

```python
f(
    "this"
    "that"
    "more"
    None,
)
```

At a glance that is four parameters, especially when looked at with longer strings of varying lengths.  Vs.

```python
f(
    (
        "this"
        "that"
        "more"
    ),
    None,
)
```

Or, depending on the scenario, something like pre-building that string into a variable, or using `.join()`, etc.